de.dlr.sc.virsat.model.concept/src/de/dlr/sc/virsat/model/concept/generator/validator/GenerateValidator.xtend

```diff
@@ -24,20 +24,20 @@ class GenerateValidator extends AGeneratorGapGenerator<EObject> {
 		concept.name + "." + PACKAGE_FOLDER;
 	}
 	
-	static def getConcreteClassName(EObject typeDefinition) {
-		"StructuralElementInstanceValidator";
+	static def getConcreteClassName(Concept concept) {
+		concept.name.replace("de.dlr.sc.virsat.model.extension.", "") + "Validator";

```

Comment:
Two things here:
1. There are now several places where we need to extract the concept name from the FQN. This should be refactored into some utility class or similar. With xtend it should allow a simple call here.
2. The current code will create invalid class names since it does not obey that the first letter of a class should be written in upper case. 

-> Please consider to externalize this code and also provide a test for correctly extracting the class name.